lib/utils/note-utils.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Matches the title and excerpt in a note's content
+ *
+ * Both the title and the excerpt are determined as
+ * content starting with something that isn't
+ * whitespace and leads up to a newline or line end
+ *
+ * @type {RegExp} matches a title and excerpt in note content
+ */
+const noteTitleAndPreviewRegExp = /\s*(\S[^\n]*)\s*(\S[^\n]*)?/;
+
+/**
+ * Matches the title and excerpt in a note's content skipping opening # from title
+ *
+ * Both the title and the excerpt are determined as
+ * content starting with something that isn't
+ * whitespace and leads up to a newline or line end
```

Comment:
nitpick: we should have a newline here. I wouldn't worry about it though - if you want to you can add it in but otherwise we'll come back later and add it. I don't want this to hold-up the PR